src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -96,5 +86,263 @@ public bool NeedsRefresh()
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.String.ToString());
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.EnabledJsonPropertyName:
+                            {
+                                if (reader.Read() && (reader.TokenType == JsonTokenType.False || reader.TokenType == JsonTokenType.True))
+                                {
+                                    featureFlag.Enabled = reader.GetBoolean();
+                                }
+                                else if (reader.TokenType == JsonTokenType.String && bool.TryParse(reader.GetString(), out bool enabled))
+                                {
+                                    featureFlag.Enabled = enabled;
+                                }
+                                else
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.EnabledJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        $"{JsonTokenType.True}' or '{JsonTokenType.False}");
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.ConditionsJsonPropertyName:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.StartObject)
+                                {
+                                    featureFlag.Conditions = ParseFeatureConditions(ref reader, settingKey);
+                                }
+                                else if (reader.TokenType == JsonTokenType.Null)
+                                {
+                                    break;
+                                }
+                                else
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.ConditionsJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.StartObject.ToString());
+                                }
+
+                                break;
+                            }
+
+                        default:
+                            reader.Skip();
+
+                            if (reader.TokenType == JsonTokenType.StartObject || reader.TokenType == JsonTokenType.StartArray)
```

Comment:
I'm a little bit confused why this is needed. Skip will position to the end of object or array at worst, and the `reader.Read()` that follows in the `while` clause should pass it.

Reply:
Since the current token is a PropertyName, Skip will move to the property value.

Reply:
Oh I didn't see the (originally, or after advancing) part, so I guess it already does that skip. I can remove this.